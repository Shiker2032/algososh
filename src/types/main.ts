import { ElementStates } from "./element-states";

export interface IString {
  letter: string;
  state: ElementStates;
}

export interface ISort {
  value: number;
  state: ElementStates;
}

export interface IStack<T> {
  push: (item: T) => void;
  pop: () => void;
  elements: () => (T | null)[];
  clear: () => void;
}

export interface IQueue<T> {
  enqueue: (data: string) => void;
  dequeue: () => void;
}

export interface IQueueElement {
  value: string;
  state: ElementStates;
}
