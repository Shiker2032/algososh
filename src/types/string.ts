import { ElementStates } from "./element-states";

export interface IString {
  letter: string;
  state: ElementStates;
}
