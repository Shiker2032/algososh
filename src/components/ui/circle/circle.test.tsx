import { render, screen } from "@testing-library/react"
import { ElementStates } from "../../../types/element-states"
import { Circle } from "./circle"

describe("Circle ui component tests", () => {
  test("circle no letter", () => {
    render(<Circle />)
    expect(screen.getByTestId("letter p")).toHaveTextContent("")
  })

  test("circle with letter", () => {
    render(<Circle letter={"test"} />)
    expect(screen.getByTestId("letter p")).toMatchSnapshot()
  })

  test("circle with head", () => {
    render(<Circle head={"test"} />)
    expect(screen.getByTestId("head div")).toMatchSnapshot()
  })

  test("circle with react element in head", () => {
    render(<Circle head={<Circle letter={"test"} />} />)
    const element = screen.getAllByTestId("head div").find((el) => el.textContent === "test")
    expect(element).toHaveClass("head")
  })

  test("circle with tail", () => {
    render(<Circle tail={"test"} />)
    expect(screen.getByTestId("tail div")).toHaveTextContent("test")
  })

  test("circle with react element in tail", () => {
    render(<Circle tail={<Circle letter={"test"} />} />)
    const element = screen.getAllByTestId("tail div").find((el) => el.textContent === "test")
    expect(element).toHaveClass("tail30")
  })

  test("circle with index", () => {
    render(<Circle index={1} />)
    expect(screen.getByTestId("index p")).toHaveTextContent("1")
  })

  test("circle with isSmall prop", () => {
    render(<Circle isSmall={true} />)
    expect(screen.getByTestId("circle div")).toHaveClass("small")
  })

  test("circle in default state", () => {
    render(<Circle />)
    expect(screen.getByTestId("circle div")).toHaveClass("default")
  })

  test("circle in changing state", () => {
    render(<Circle state={ElementStates.Changing} />)
    expect(screen.getByTestId("circle div")).toHaveClass("changing")
  })

  test("circle in modified state", () => {
    render(<Circle state={ElementStates.Modified} />)
    expect(screen.getByTestId("circle div")).toHaveClass("modified")
  })
})
