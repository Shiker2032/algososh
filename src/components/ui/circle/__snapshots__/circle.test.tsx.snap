// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Circle component tests circle with head 1`] = `
<div
  class="text text_type_input text_color_input mb-4 absolute head string"
  data-testid="head div"
>
  test
</div>
`;

exports[`Circle component tests circle with letter 1`] = `
<p
  class="text text_type_circle text_color_input letter"
  data-testid="letter p"
>
  test
</p>
`;
